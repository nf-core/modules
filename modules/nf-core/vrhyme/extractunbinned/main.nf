process VRHYME_EXTRACTUNBINNED {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/vrhyme:1.1.0--pyhdfd78af_1':
        'biocontainers/vrhyme:1.1.0--pyhdfd78af_1' }"

    input:
    tuple val(meta), path(membership)
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*_unbinned_sequences.fasta") , emit: unbinned_sequences
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_unbinned_sequences.py \\
        -i $membership \\
        -f $fasta \\
        -o ${prefix}_unbinned_sequences.fasta \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vrhyme: \$(echo \$(vRhyme --version 2>&1) | sed 's/^.*vRhyme v//; s/Using.*\$//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_unbinned_sequences.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vrhyme: \$(echo \$(vRhyme --version 2>&1) | sed 's/^.*vRhyme v//; s/Using.*\$//' )
    END_VERSIONS
    """
}
