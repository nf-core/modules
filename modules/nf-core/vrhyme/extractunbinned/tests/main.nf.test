nextflow_process {

    name "Test Process VRHYME_EXTRACTUNBINNED"

    script "../main.nf"
    process "VRHYME_EXTRACTUNBINNED"

    tag "modules"
    tag "modules_nfcore"
    tag "vrhyme"
    tag "vrhyme/vrhyme"
    tag "vrhyme/extractunbinned"

    setup {

        run("VRHYME_VRHYME") {
            script "../../vrhyme/main.nf"

            process {
                """
                input[0] = [
                    [id:'test',single_end:false],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                input[1] = [
                    [id:'test',single_end:false],// meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                    ]
                """
            }
        }
    }

    test("test_vrhyme_extractunbinned") {

        when {
            process {
                """
                input[0] = VRHYME_VRHYME.out.membership
                input[1] = [
                    [id:'test',single_end:false],// meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("test_vrhyme_extractunbinned -- stub") {
        options '-stub'
        when {
            process {
                """
                input[0] = VRHYME_VRHYME.out.membership
                input[1] = [
                    [id:'test',single_end:false],// meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                    ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}