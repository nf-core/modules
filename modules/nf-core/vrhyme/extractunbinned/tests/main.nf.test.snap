{
    "test_vrhyme_extractunbinned -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_unbinned_sequences.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,19f257878bced6c770eaf8ed45151b96"
                ],
                "unbinned_sequences": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_unbinned_sequences.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,19f257878bced6c770eaf8ed45151b96"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-30T20:31:33.979555851"
    },
    "test_vrhyme_extractunbinned": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_unbinned_sequences.fasta:md5,816e986c8ad8f277eb4a4891a12d7aff"
                    ]
                ],
                "1": [
                    "versions.yml:md5,19f257878bced6c770eaf8ed45151b96"
                ],
                "unbinned_sequences": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_unbinned_sequences.fasta:md5,816e986c8ad8f277eb4a4891a12d7aff"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,19f257878bced6c770eaf8ed45151b96"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-31T10:00:14.099653894"
    }
}